pykusto/expressions.py

```diff
@@ -238,8 +238,8 @@ def concat(*strings: StringType) -> 'StringExpression':
 
     def split(self, delimiter: StringType, requested_index: NumberType = None) -> 'ArrayExpression':
         if requested_index is None:
-            return ArrayExpression(KQL('split({}, {}'.format(self.kql, delimiter)))
-        return ArrayExpression(KQL('split({}, {}, {}'.format(self.kql, delimiter, requested_index)))
+            return ArrayExpression(KQL('split({}, "{}")'.format(_subexpr_to_kql(self.kql), delimiter)))
+        return ArrayExpression(KQL('split({}, "{}", {})'.format(_subexpr_to_kql(self.kql), delimiter, requested_index)))
```

Comment:
The correct way to generate enclosing quotes is to call `_subexpr_to_kql` on the arguments